pkg/operator/reconciler.go

```diff
@@ -0,0 +1,246 @@
+package operator
+
+import (
+	"context"
+	"os"
+	"reflect"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	errors "k8s.io/apimachinery/pkg/api/errors"
+	k8serros "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	tunedv1 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/tuned/v1"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/manifests"
+	ntomf "github.com/openshift/cluster-node-tuning-operator/pkg/manifests"
+)
+
+// Reconciler reconciles a Tuned object and creates the tuned daemon set.
+type Reconciler struct {
+	Client    client.Client
+	Scheme    *runtime.Scheme
+	Namespace string
+}
+
+// SetupWithManager creates a new ClusterOperator Controller and adds it to the Manager.
+// The Manager will set fields on the Controller and Start it when the Manager is Started.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// We want to initiate reconcile loop only on changes under the NTO cluster operator.
+	cvoPredicates := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			if e.Object == nil {
+				klog.Error("create event has no runtime object to create")
+				return false
+			}
+
+			return e.Object.GetName() == tunedv1.TunedClusterOperatorResourceName
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if !validateUpdateEvent(&e) {
```

Comment:
Are you doing it to address the scenarios that the reconciliation fails because the resource changed on the cluster? If yes, I'd suggest using the client and fetching the resource that you want to change/update before calling the update. Then, if it fails return err in the reconciliation to ensure that it will be executed again.